### PR TITLE
Publish Imagemosaic Worker general improvements

### PR DESCRIPTION
**Pull request by hblitza**
Besides minors this adds:
- time check and enable time dimension for the layer. I think this the time definition will be changed as soon as we have proper data
- option to replace existing granules (fourth input param)

TimeDimension Code based of Sauber project :pray:

please have a look: @weskamm @chrismayer

### CODE REVIEW

File: src/geoserver-publish-imagemosaic/index.js [hunk 1]
```diff
@@ -22,61 +22,91 @@ const grc = new GeoServerRestClient(url, user, pw);
...
     );
     log('Successfully added new granule to coverage store.');
+
+    // check if layer has time dimension enabled
+    let hasTime = false;
+    const coverage = await grc.layers.getCoverage(ws, covStore, covStore);
+    if (coverage && coverage.coverage.metadata && coverage.coverage.metadata.entry &&
+      coverage.coverage.metadata.entry['@key'] === 'time' && (typeof coverage.coverage.metadata.entry.dimensionInfo === 'object')) {
+      const dimInfo = coverage.coverage.metadata.entry.dimensionInfo;
+      if (dimInfo.enabled === true && dimInfo.nearestMatchEnabled === true &&
```
Comment (weskamm) [inline src/geoserver-publish-imagemosaic/index.js:95]: i think having nearest match enabled is optional and does not need to be checked

File: src/geoserver-publish-imagemosaic/index.js [hunk 2]
```diff
@@ -22,61 +22,91 @@ const grc = new GeoServerRestClient(url, user, pw);
...
+        dimInfo.acceptableInterval) {
+        hasTime = true;
+      }
+    }
+
+    if (!hasTime) {
+      console.info(`Enabling time for layer "${ws}:${layerName}"`);
```
Comment (weskamm) [inline src/geoserver-publish-imagemosaic/index.js:102]: please use log function

**Review by weskamm — APPROVED** [review]
found 2 small minors, looks good!

**Comment by hblitza** [comment]
Thanks!

**Pull request merged**